README.md
## NOTE: This is a fork of [gruvbox_material](https://github.com/sainnhe/gruvbox-material). I just changed some colors to my liking. The changed colors will only be effective if `background=dark` and `g:gruvbox_material_background='hard'`

-   changed background color from `#1d2021` to `#1c1c1c`.
    <!-- ./autoload/gruvbox-material.vim#L6 -->
-   changed tabline fill color from `#282828` to `#1c1c1c` (for airline)
    <!-- ./autoload/gruvbox_material.vim#L118 -->
    <!-- ./autoload/airline/themes/gruvbox_material.vim#L101 -->
-   changed inactive tab color from `#504945` to `#3c3836` (for airline)
    <!-- ./autoload/gruvbox_material.vim#L116 -->
    <!-- ./autoload/airline/themes/gruvbox_material.vim#L156 -->
-   removed `fzf.vim` color scheme. Bring your own config or use the default one below.
    <!-- ./autoload/gruvbox_material.vim#L495-L508 -->

```vim
let g:fzf_colors = {
      \ 'fg':      ['fg', 'Normal'],
      \ 'bg':      ['bg', 'Normal'],
      \ 'hl':      ['fg', 'Green'],
      \ 'fg+':     ['fg', 'CursorLine', 'CursorColumn', 'Normal'],
      \ 'bg+':     ['bg', 'CursorLine', 'CursorColumn'],
      \ 'hl+':     ['fg', 'Aqua'],
      \ 'info':    ['fg', 'Aqua'],
      \ 'prompt':  ['fg', 'Orange'],
      \ 'pointer': ['fg', 'Blue'],
      \ 'marker':  ['fg', 'Yellow'],
      \ 'spinner': ['fg', 'Yellow'],
      \ 'header':  ['fg', 'Grey']
      \ }
```

## Introduction

Gruvbox Material is a modified version of [Gruvbox](https://github.com/morhetz/gruvbox), the contrast is adjusted to be softer in order to protect developers' eyes.

There are 3 palettes available in this color scheme:

<details>
  <summary><code>material</code>: Carefully designed to have a soft contrast(<b>click on this line to preview</b>)</summary>

|        |                                                             𝐃𝐚𝐫𝐤                                                              |                                                             𝐋𝐢𝐠𝐡𝐭                                                              |
| :----: | :---------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------------------------------------: |
|  𝐇𝐚𝐫𝐝  |  ![material-hard-dark](https://user-images.githubusercontent.com/37491630/75227134-891fbb80-57a5-11ea-878e-b8b2972cfd6e.png)  |  ![material-hard-light](https://user-images.githubusercontent.com/37491630/75227137-8a50e880-57a5-11ea-90dc-b2646d8b0b55.png)  |
| 𝐌𝐞𝐝𝐢𝐮𝐦 | ![material-medium-dark](https://user-images.githubusercontent.com/37491630/75227139-8cb34280-57a5-11ea-86d6-3d3f6a2475eb.png) | ![material-medium-light](https://user-images.githubusercontent.com/37491630/75227141-8de46f80-57a5-11ea-820a-9394ab9d09aa.png) |
|  𝐒𝐨𝐟𝐭  |  ![material-soft-dark](https://user-images.githubusercontent.com/37491630/75227149-9046c980-57a5-11ea-8633-bf4f31e533d0.png)  |  ![material-soft-light](https://user-images.githubusercontent.com/37491630/75227157-92108d00-57a5-11ea-8b13-b2130bff60d8.png)  |

</details>

<details>
  <summary><code>mix</code>: Color palette obtained by calculating the mean of the other two(<b>click on this line to preview</b>)</summary>

|        |                                                           𝐃𝐚𝐫𝐤                                                           |                                                           𝐋𝐢𝐠𝐡𝐭                                                           |
| :----: | :----------------------------------------------------------------------------------------------------------------------: | :-----------------------------------------------------------------------------------------------------------------------: |
|  𝐇𝐚𝐫𝐝  |  ![mix-hard-dark](https://user-images.githubusercontent.com/37491630/76383368-826f7780-6353-11ea-8094-b593eb5f1e10.png)  |  ![mix-hard-light](https://user-images.githubusercontent.com/37491630/76383372-88655880-6353-11ea-9441-78d159600faf.png)  |
| 𝐌𝐞𝐝𝐢𝐮𝐦 | ![mix-medium-dark](https://user-images.githubusercontent.com/37491630/76383370-84393b00-6353-11ea-88de-804a781d3142.png) | ![mix-medium-light](https://user-images.githubusercontent.com/37491630/76383375-8ac7b280-6353-11ea-94a8-62e3845203bc.png) |
|  𝐒𝐨𝐟𝐭  |  ![mix-soft-dark](https://user-images.githubusercontent.com/37491630/76383371-869b9500-6353-11ea-923d-9011bbe6bcad.png)  |  ![mix-soft-light](https://user-images.githubusercontent.com/37491630/76383380-8c917600-6353-11ea-8530-a67932a6a2ec.png)  |

</details>

<details>
  <summary><code>original</code>: The color palette used in the original gruvbox(<b>click on this line to preview</b>)</summary>

|        |                                                             𝐃𝐚𝐫𝐤                                                              |                                                             𝐋𝐢𝐠𝐡𝐭                                                              |
| :----: | :---------------------------------------------------------------------------------------------------------------------------: | :----------------------------------------------------------------------------------------------------------------------------: |
|  𝐇𝐚𝐫𝐝  |  ![original-hard-dark](https://user-images.githubusercontent.com/37491630/76383382-8e5b3980-6353-11ea-9398-08d31b1ed32d.png)  |  ![original-hard-light](https://user-images.githubusercontent.com/37491630/76383389-931fed80-6353-11ea-905f-47b35c0cac39.png)  |
| 𝐌𝐞𝐝𝐢𝐮𝐦 | ![original-medium-dark](https://user-images.githubusercontent.com/37491630/76383385-9024fd00-6353-11ea-99c1-7bba4f796115.png) | ![original-medium-light](https://user-images.githubusercontent.com/37491630/76383393-94511a80-6353-11ea-84ea-551b44f0d5bd.png) |
|  𝐒𝐨𝐟𝐭  |  ![original-soft-dark](https://user-images.githubusercontent.com/37491630/76383387-91562a00-6353-11ea-90a0-daac8653dfd0.png)  |  ![original-soft-light](https://user-images.githubusercontent.com/37491630/76383396-95824780-6353-11ea-9b36-302b88fef429.png)  |

</details>

This color scheme uses the `material` palette by default, you can use a global variable to switch to another palette.

### Features

-   Carefully designed color palette for eye protection.
-   Switch back to the original color palette whenever you like.
-   Highly customizable.
-   Rich support for common file types and plugins.
-   [Italic support](https://github.com/sainnhe/icursive-nerd-font) 🎉

## Installation

### Via Plugin Manager

Take [vim-plug](https://github.com/junegunn/vim-plug) for example:

```vim
Plug 'numtostr/gruvbox-material'
```

For better syntax highlighting support, please install [sheerun/vim-polyglot](https://github.com/sheerun/vim-polyglot) or [nvim-treesitter/nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter).

### Manually

1. Clone this repository.
2. Copy `/path/to/gruvbox-material/autoload/gruvbox_material.vim` to `~/.vim/autoload/`.
3. Copy `/path/to/gruvbox-material/colors/gruvbox-material.vim` to `~/.vim/colors/` .
4. Copy `/path/to/gruvbox-material/doc/gruvbox-material.txt` to `~/.vim/doc/` and execute `:helptags ~/.vim/doc/` to generate help tags.
5. To install [airline](https://github.com/vim-airline/vim-airline) theme, copy `/path/to/gruvbox-material/autoload/airline/themes/gruvbox_material.vim` to `~/.vim/autoload/airline/themes/` .
6. To install [lightline](https://github.com/itchyny/lightline.vim) theme, copy `/path/to/gruvbox-material/autoload/lightline/colorscheme/gruvbox_material.vim` to `~/.vim/autoload/lightline/colorscheme/` .

### AUR

There are some packages available for Arch Linux users in AUR:

-   [vim-gruvbox-material-git](https://aur.archlinux.org/pkgbase/vim-gruvbox-material-git/)
-   [neovim-gruvbox-material-git](https://aur.archlinux.org/pkgbase/neovim-gruvbox-material-git/)

## Usage

See [`:help gruvbox-material.txt`](https://github.com/sainnhe/gruvbox-material/blob/master/doc/gruvbox-material.txt)

## Related Projects

See this [wiki page](https://github.com/sainnhe/gruvbox-material/wiki/Related-Projects).

## More Color Schemes

-   [Everforest](https://github.com/sainnhe/everforest)
-   [Edge](https://github.com/sainnhe/edge)
-   [Sonokai](https://github.com/sainnhe/sonokai)

## Credits

-   The color palette is based on [gruvbox](https://github.com/morhetz/gruvbox) by [@morhetz](https://github.com/morhetz)
-   The color palette is adjusted by [@sainnhe](https://github.com/sainnhe)

## Inspirations

-   [morhetz/gruvbox](https://github.com/morhetz/gruvbox): original gruvbox
-   [atom.io](https://atom.io): the `white` color in the dark variant
-   [zefei/cake16](https://github.com/zefei/cake16): the `black` color in the light variant
-   [Google Material Design](https://www.material.io)

## Maintainers

| [![Sainnhe Park](https://avatars1.githubusercontent.com/u/37491630?s=70&u=14e72916dcf467f393c532536387ec72a23747ec&v=4)](https://github.com/sainnhe) | [![Zixin Yin](https://avatars2.githubusercontent.com/u/33487417?s=70&u=c6acee32fad2edb8c3bc3f7e0f436f1e8b8024c6&v=4)](https://github.com/zxYin) |
| ---------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------------------------------------------------------------------------------------------------------------------------- |
| [Sainnhe Park](https://github.com/sainnhe)                                                                                                           | [Zixin Yin](https://github.com/zxYin)                                                                                                           |

## License

[MIT](./LICENSE) © sainnhe


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file.

## [Unreleased]

### Added

-   Optimize `lervag/vimtex`.
-   Add `g:gruvbox_material_diagnostic_text_highlight`
-   Optimize `liuchengxu/vim-clap`.
-   Add whitelist for extended file types.
-   Support for leaderf popup mode.
-   Optimize `nvim-telescope/telescope.nvim`.

### Changed

-   Diagnostics text highlights.
-   Adjust diff and visual background colors in light variant.

## [1.2.1] - 2020-12-29

### Added

-   Optimize `nvim-treesitter/nvim-treesitter`.
-   AUR packages.
-   Support neovim lsp diagnostics.
-   New highlight groups for coc-explorer.
-   Outline of help doc.

### Changed

-   Remove duplicate docs in README.

### Fixed

-   Fix errors when doesn't set `g:colors_name`.
-   Fix `DiffText` in Vim.
-   Fix `CurrentWord` in diff mode.
-   Fix `CursorColumn` in diff mode.
-   Clear highlights when better performance is disabled.

### Changed

-   Remove duplicate docs in README.

## [1.2.0] - 2020-06-04

### Added

-   Add `g:gruvbox_material_diagnostic_line_highlight`.
-   Add `g:gruvbox_material_better_performance`.

### Changed

-   Fix airline warning and error.
-   Fix `g:gruvbox_material_current_word` settings.
-   Use `grey0` in `Conceal`.

## [1.1.3] - 2020-05-06

### Added

-   Optimize Coc.
-   Optimize gitcommit.
-   Add hi groups for builtin lsp and checkhealth.
-   Optimize DOS INI.
-   Optimize Matlab.
-   Optimize Airline.
-   Optimize `cohama/agit.vim`.
-   Add `ToolbarLine` and `ToolbarButton`.
-   Support command line mode for airline and lightline.
-   Support terminal mode for airline and lightline.
-   Add `g:gruvbox_material_statusline_style`.
-   Add more available values for `g:gruvbox_material_menu_selection_background`.
-   Add `Substitute`, `TermCursor` and `TermCursorNC`.
-   Optimize vim-sneak.
-   Add `g:gruvbox_material_sign_column_background`.

### Changed

-   Adjust `Search` and `IncSearch`.
-   Adjust `NonText`, `Whitespace` and `SpecialKey`.
-   Use green in modified fg.
-   Add `autoload/gruvbox_material.vim`.
-   Adjust `bg_red` in the mix palette.
-   Adjust `DiffText`.
-   Adjust `bg_statusline3` in soft dark background.
-   Optimize performance.

### Removed

-   Remove `g:gruvbox_material_cursor_line_contrast`.

## [1.1.2] - 2020-04-04

### Added

-   Add `g:gruvbox_material_cursor_line_contrast`
-   Optimize [arakashic/chromatica.nvim](https://github.com/arakashic/chromatica.nvim)
-   Optimize [numirias/semshi](https://github.com/numirias/semshi)
-   Optimize [jackguo380/vim-lsp-cxx-highlight](https://github.com/jackguo380/vim-lsp-cxx-highlight)

## [1.1.1] - 2020-03-11

### Changed

-   Rename `median` as `mix`.

## [1.1.0] - 2020-03-11

### Added

-   Add `g:gruvbox_material_cursor`.
-   Optimize `Yggdroot/indentLine`.
-   Add `median` palette.

### Changed

-   Adjust some background colors used in the light variant.
-   Transfer ownership.

### Fixed

-   Fix terminal colors. [#24](https://github.com/sainnhe/gruvbox-material/issues/24)

## [1.0.1] - 2020-01-09

### Added

-   Add `g:gruvbox_material_palette`.
-   Add `g:gruvbox_material_enable_italic`.
-   Add `g:gruvbox_material_menu_selection_background`.

### Changed

-   Refactor `colors/gruvbox-material.vim`.
-   Adjust syntax highlighting logic.
-   Remove color templates.
-   Rewrite airline theme.
-   Markdown heading delimiter.

## [1.0.0] - 2019-12-11

### Changed

-   Adjust the background color of `CursorLine` in the dark variant.
-   Link `WhichKeyFloating` to `SignColumn`.
-   Optimize vim-sneak.
-   Markdown headings.

### Added

-   Use blue as foreground color of modified file name in airline color scheme.
-   Add a judgment logic for `CursorLineNr`.
-   Optimize Dart.

### Fix

-   Fix background colors in transparent mode.
-   Remove `doc/tags`.

## [0.3.4] - 2019-11-23

### Changed

-   Use grey as foreground color of `Conceal`.
-   Optimize `CursorLine`.
-   Optimize `MatchParen`.
-   Optimize `Pmenu`.
-   Better support for hard light variant.
-   Disable bold in `MatchParen`.

### Added

-   Support `Shougo/denite.nvim`.
-   Support `mg979/vim-visual-multi`.
-   Support `liuchengxu/vim-which-key`.
-   Support `neomake/neomake`.
-   Support `liuchengxu/vista.vim`.
-   Support `unblevable/quick-scope`.
-   Support `neoclide/coc-yank`.
-   New configuration option: `g:gruvbox_material_visual`.
-   New configuration option: `g:gruvbox_material_current_word`.

## [0.3.3] - 2019-11-07

### Changed

-   Adjust white in the dark variant.

## [0.3.2] - 2019-11-01

### Changed

-   Adjust white in the dark variant.

## [0.3.1] - 2019-10-31

### Changed

-   Adjust white in the dark variant.

## [0.3.0] - 2019-10-30

### Changed

-   Adjust yellow in the dark variant.
-   Adjust red, orange and blue in the light variant.

### Added

-   Add `CHANGELOG.md`.

## [0.2.0] - 2019-10-25

### Changed

-   Adjust white and yellow in the dark variant.
-   Adjust black in the light variant.

## [0.1.4] - 2019-10-17

### Fixed

-   Stable release for the initial color palette.

## [0.1.3] - 2019-09-22

### Fixed

-   Better support for `:set notermguicolors`.

## [0.1.2] - 2019-09-16

### Changed

-   Adjust workbench colors.

## [0.1.1] - 2019-09-04

### Changed

-   Adjust the color palette of the light variant.

## [0.1.0] - 2019-09-03

### Added

-   Finish light variant.

## [0.0.3] - 2019-07-24

### Added

-   Optimize markdown.
-   Optimize ALE Virtual Text, CoC and vim-matchup.
-   New option: `g:gruvbox_material_lightline_kill_bold`.

### Changed

-   Adjust blue in order to better distinguish between blue and aqua.

### Fixed

-   Foreground color of parentheses.

## [0.0.2] - 2019-07-23

### Changed

-   Adjust blue in order to better distinguish between blue and aqua.
-   Adjust the location of hi links for better load time.

### Fixed

-   Fix terminal colors.

## [0.0.1] - 2019-07-22

### Added

-   Finish dark variant.
-   Airline theme.
-   Lightline theme.

lua/gruvbox-material/init.lua
local definitions = require("gruvbox-material.definitions")
local api = vim.api
local highlight = api.nvim_set_hl
local cmd = api.nvim_command

local gruvbox = {
    defaults = {
        enable = true, -- boolean
        background = "hard", -- 'hard' | 'medium' | 'soft'
        palette = "material" -- 'material' | 'mix' | 'original'
    }
}

cmd("hi clear")

local function set_namespace(ns_name)
    -- Color namespace
    local ns = api.nvim_create_namespace(ns_name)

    -- Activate namespace

    -- This API isn't stable yet. It will receive breaking changes
    -- and be renamed to nvim_set_hl_ns later be aware of that.
    -- for more details https://github.com/neovim/neovim/issues/14090#issuecomment-799285918
    api.nvim__set_hl_ns(ns)

    return ns
end

function gruvbox.setup(opts)
    opts = opts ~= nil and vim.tbl_extend("keep", gruvbox.defaults, opts) or gruvbox.defaults

    -- Color namespace
    local ns = set_namespace("gruvbox-material")
    local defs = definitions.get_definitions(opts)

    -- Set highlights
    for color_name, param in pairs(defs) do
        highlight(ns, color_name, param)
    end

    if opts.enable then
        cmd("colorscheme gruvbox-material")
    end
end

return gruvbox


lua/gruvbox-material/palette.lua
local palette = {}
local bg = vim.o.background
local rgb_color = vim.api.nvim_get_color_by_name

local is_dark = bg == "dark"

local function palette_1(gruvbox_bg)
    if gruvbox_bg == "hard" then
        if is_dark then
            return {
                bg0 = {"#1a1a1a", "234"},
                bg1 = {"#282828", "235"},
                bg2 = {"#282828", "235"},
                bg3 = {"#3c3836", "237"},
                bg4 = {"#3c3836", "237"},
                bg5 = {"#504945", "239"},
                bg_statusline1 = {"#282828", "235"},
                bg_statusline2 = {"#32302f", "235"},
                bg_statusline3 = {"#504945", "239"},
                bg_diff_green = {"#32361a", "22"},
                bg_visual_green = {"#333e34", "22"},
                bg_diff_red = {"#3c1f1e", "52"},
                bg_visual_red = {"#442e2d", "52"},
                bg_diff_blue = {"#0d3138", "17"},
                bg_visual_blue = {"#2e3b3b", "17"},
                bg_visual_yellow = {"#473c29", "94"},
                bg_current_word = {"#32302f", "236"}
            }
        else
            return {
                bg0 = {"#f9f5d7", "230"},
                bg1 = {"#f5edca", "229"},
                bg2 = {"#f3eac7", "229"},
                bg3 = {"#f2e5bc", "228"},
                bg4 = {"#eee0b7", "223"},
                bg5 = {"#ebdbb2", "223"},
                bg_statusline1 = {"#f5edca", "223"},
                bg_statusline2 = {"#f3eac7", "223"},
                bg_statusline3 = {"#eee0b7", "250"},
                bg_diff_green = {"#e4edc8", "194"},
                bg_visual_green = {"#dde5c2", "194"},
                bg_diff_red = {"#f8e4c9", "217"},
                bg_visual_red = {"#f0ddc3", "217"},
                bg_diff_blue = {"#e0e9d3", "117"},
                bg_visual_blue = {"#d9e1cc", "117"},
                bg_visual_yellow = {"#f9eabf", "226"},
                bg_current_word = {"#f3eac7", "229"}
            }
        end
    elseif gruvbox_bg == "medium" then
        if is_dark then
            return {
                bg0 = {"#282828", "235"},
                bg1 = {"#32302f", "236"},
                bg2 = {"#32302f", "236"},
                bg3 = {"#45403d", "237"},
                bg4 = {"#45403d", "237"},
                bg5 = {"#5a524c", "239"},
                bg_statusline1 = {"#32302f", "236"},
                bg_statusline2 = {"#3a3735", "236"},
                bg_statusline3 = {"#504945", "240"},
                bg_diff_green = {"#34381b", "22"},
                bg_visual_green = {"#3b4439", "22"},
                bg_diff_red = {"#402120", "52"},
                bg_visual_red = {"#4c3432", "52"},
                bg_diff_blue = {"#0e363e", "17"},
                bg_visual_blue = {"#374141", "17"},
                bg_visual_yellow = {"#4f422e", "94"},
                bg_current_word = {"#3c3836", "237"}
            }
        else
            return {
                bg0 = {"#fbf1c7", "229"},
                bg1 = {"#f4e8be", "228"},
                bg2 = {"#f2e5bc", "228"},
                bg3 = {"#eee0b7", "223"},
                bg4 = {"#e5d5ad", "223"},
                bg5 = {"#ddccab", "250"},
                bg_statusline1 = {"#f2e5bc", "223"},
                bg_statusline2 = {"#f2e5bc", "223"},
                bg_statusline3 = {"#e5d5ad", "250"},
                bg_diff_green = {"#e6eabc", "194"},
                bg_visual_green = {"#dee2b6", "194"},
                bg_diff_red = {"#f9e0bb", "217"},
                bg_visual_red = {"#f1d9b5", "217"},
                bg_diff_blue = {"#e2e6c7", "117"},
                bg_visual_blue = {"#dadec0", "117"},
                bg_visual_yellow = {"#fae7b3", "226"},
                bg_current_word = {"#f2e5bc", "228"}
            }
        end
    elseif gruvbox_bg == "soft" then
        if is_dark then
            return {
                bg0 = {"#32302f", "236"},
                bg1 = {"#3c3836", "237"},
                bg2 = {"#3c3836", "237"},
                bg3 = {"#504945", "239"},
                bg4 = {"#504945", "239"},
                bg5 = {"#665c54", "241"},
                bg_statusline1 = {"#3c3836", "237"},
                bg_statusline2 = {"#46413e", "237"},
                bg_statusline3 = {"#5b534d", "241"},
                bg_diff_green = {"#3d4220", "22"},
                bg_visual_green = {"#424a3e", "22"},
                bg_diff_red = {"#472322", "52"},
                bg_visual_red = {"#543937", "52"},
                bg_diff_blue = {"#0f3a42", "17"},
                bg_visual_blue = {"#404946", "17"},
                bg_visual_yellow = {"#574833", "94"},
                bg_current_word = {"#45403d", "238"}
            }
        else
            return {
                bg0 = {"#f2e5bc", "228"},
                bg1 = {"#eddeb5", "223"},
                bg2 = {"#ebdbb2", "228"},
                bg3 = {"#e6d5ae", "223"},
                bg4 = {"#dac9a5", "250"},
                bg5 = {"#d5c4a1", "250"},
                bg_statusline1 = {"#ebdbb2", "223"},
                bg_statusline2 = {"#ebdbb2", "223"},
                bg_statusline3 = {"#dac9a5", "250"},
                bg_diff_green = {"#dfe1b4", "194"},
                bg_visual_green = {"#d7d9ae", "194"},
                bg_diff_red = {"#f7d9b9", "217"},
                bg_visual_red = {"#efd2b3", "217"},
                bg_diff_blue = {"#dbddbf", "117"},
                bg_visual_blue = {"#d3d5b8", "117"},
                bg_visual_yellow = {"#f3deaa", "226"},
                bg_current_word = {"#ebdbb2", "227"}
            }
        end
    end
end

local function palette_2(gruvbox_palette)
    if gruvbox_palette == "material" then
        if is_dark then
            return {
                fg0 = {"#d4be98", "223"},
                fg1 = {"#ddc7a1", "223"},
                red = {"#ea6962", "167"},
                orange = {"#e78a4e", "208"},
                yellow = {"#d8a657", "214"},
                green = {"#a9b665", "142"},
                aqua = {"#89b482", "108"},
                blue = {"#7daea3", "109"},
                purple = {"#d3869b", "175"},
                bg_red = {"#ea6962", "167"},
                bg_green = {"#a9b665", "142"},
                bg_yellow = {"#d8a657", "214"}
            }
        else
            return {
                fg0 = {"#654735", "237"},
                fg1 = {"#4f3829", "237"},
                red = {"#c14a4a", "88"},
                orange = {"#c35e0a", "130"},
                yellow = {"#b47109", "136"},
                green = {"#6c782e", "100"},
                aqua = {"#4c7a5d", "165"},
                blue = {"#45707a", "24"},
                purple = {"#945e80", "96"},
                bg_red = {"#ae5858", "88"},
                bg_green = {"#6f8352", "100"},
                bg_yellow = {"#a96b2c", "130"}
            }
        end
    elseif gruvbox_palette == "mix" then
        if is_dark then
            return {
                fg0 = {"#e2cca9", "223"},
                fg1 = {"#e2cca9", "223"},
                red = {"#f2594b", "167"},
                orange = {"#f28534", "208"},
                yellow = {"#e9b143", "214"},
                green = {"#b0b846", "142"},
                aqua = {"#8bba7f", "108"},
                blue = {"#80aa9e", "109"},
                purple = {"#d3869b", "175"},
                bg_red = {"#db4740", "167"},
                bg_green = {"#b0b846", "142"},
                bg_yellow = {"#e9b143", "214"}
            }
        else
            return {
                fg0 = {"#514036", "237"},
                fg1 = {"#514036", "237"},
                red = {"#af2528", "88"},
                orange = {"#b94c07", "130"},
                yellow = {"#b4730e", "136"},
                green = {"#72761e", "100"},
                aqua = {"#477a5b", "165"},
                blue = {"#266b79", "24"},
                purple = {"#924f79", "96"},
                bg_red = {"#ae5858", "88"},
                bg_green = {"#6f8352", "100"},
                bg_yellow = {"#a96b2c", "130"}
            }
        end
    elseif gruvbox_palette == "original" then
        if is_dark then
            return {
                fg0 = {"#ebdbb2", "223"},
                fg1 = {"#ebdbb2", "223"},
                red = {"#fb4934", "167"},
                orange = {"#fe8019", "208"},
                yellow = {"#fabd2f", "214"},
                green = {"#b8bb26", "142"},
                aqua = {"#8ec07c", "108"},
                blue = {"#83a598", "109"},
                purple = {"#d3869b", "175"},
                bg_red = {"#cc241d", "124"},
                bg_green = {"#b8bb26", "106"},
                bg_yellow = {"#fabd2f", "172"}
            }
        else
            return {
                fg0 = {"#3c3836", "237"},
                fg1 = {"#3c3836", "237"},
                red = {"#9d0006", "88"},
                orange = {"#af3a03", "130"},
                yellow = {"#b57614", "136"},
                green = {"#79740e", "100"},
                aqua = {"#427b58", "165"},
                blue = {"#076678", "24"},
                purple = {"#8f3f71", "96"},
                bg_red = {"#ae5858", "88"},
                bg_green = {"#6f8352", "100"},
                bg_yellow = {"#a96b2c", "130"}
            }
        end
    end
end

local function palette_3()
    if is_dark then
        return {
            grey0 = {"#7c6f64", "243"},
            grey1 = {"#928374", "245"},
            grey2 = {"#a89984", "246"}
        }
    else
        return {
            grey0 = {"#a89984", "246"},
            grey1 = {"#928374", "245"},
            grey2 = {"#7c6f64", "243"}
        }
    end
end

function palette.get_palette(gruvbox_bg, gruvbox_palette)
    local p1 = palette_1(gruvbox_bg)
    local p2 = palette_2(gruvbox_palette)
    local p3 = palette_3()

    local colors = {}

    for key, value in pairs(p1) do
        colors[key] = rgb_color(value[1])
    end

    for key, value in pairs(p2) do
        colors[key] = rgb_color(value[1])
    end

    for key, value in pairs(p3) do
        colors[key] = rgb_color(value[1])
    end

    return colors
end

return palette
